wenet/utils/class_utils.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright [2023-11-28] <sxc19@mails.tsinghua.edu.cn, Xingchen Song>
+import torch
+
+from wenet.transformer.subsampling import (
+    LinearNoSubsampling, EmbedinigNoSubsampling,
+    Conv1dSubsampling2, Conv2dSubsampling4,
+    Conv2dSubsampling6, Conv2dSubsampling8,
+)
+from wenet.efficient_conformer.subsampling import Conv2dSubsampling2
+from wenet.squeezeformer.subsampling import DepthwiseConv2dSubsampling4
+from wenet.transformer.embedding import (
+    PositionalEncoding, RelPositionalEncoding,
+    WhisperPositionalEncoding, LearnablePositionalEncoding,
+    NoPositionalEncoding
+)
+from wenet.transformer.attention import (
+    MultiHeadedAttention, RelPositionMultiHeadedAttention
+)
+from wenet.efficient_conformer.attention import GroupedRelPositionMultiHeadedAttention
+
+
+def get_activation(act):
+    """Return activation function."""
+    # Lazy load to avoid unused import
+    from wenet.transformer.swish import Swish
+
+    activation_funcs = {
+        "hardtanh": torch.nn.Hardtanh,
+        "tanh": torch.nn.Tanh,
+        "relu": torch.nn.ReLU,
+        "selu": torch.nn.SELU,
+        "swish": getattr(torch.nn, "SiLU", Swish),
+        "gelu": torch.nn.GELU
+    }
+
+    return activation_funcs[act]()
+
+
+def get_rnn(rnn_type: str) -> torch.nn.Module:
+    assert rnn_type in ["rnn", "lstm", "gru"]
+    if rnn_type == "rnn":
+        return torch.nn.RNN
+    elif rnn_type == "lstm":
+        return torch.nn.LSTM
+    else:
+        return torch.nn.GRU
+
+
+WENET_SUBSAMPLE_CLASSES = {
```

Comment:
do you think it's more clear if we follow the name patter like before, such as `get_subsamling`, `get_embedding`, and `get_attention`.

Reply:
`get_activation` return instance of class while `WENET_XXX_CLASSES` return class. To keep the name pattern, it's better to refine `get_actiavtion` to `WENET_ACTIVATION_CLASSES`, because we can not get  instance of subsampling class without arguments

Reply:
ok